### PR TITLE
[Admin][UI] Unified ChannelCollection template

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12          |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| Related tickets | continuation of #12119                      |
| License         | MIT                                                          |

### Changes:
- Fixed the semantic's tab mechanism being unhooked in dynamic forms (for instance, changing shipping charges from `per shipment` to `per unit` would result in the tabs not working at all)
- Fixed switching tabs behavior when more than one channel collection form is on the page (due to the tabs being previously connected with their content by channel's code, changing a tab in one form would cascade to all others on the page)
- Added a visible error indicator to the tabs showing in which channel and how many errors are present along with behats for that <br>image

### Possible future improvements:
- Adding `required` asterisks to channel tabs when the underlying form has required fields as discussed in the original PR
- Removing the grouped validation error from product variant forms (**You must define price for every channel** etc), and substituting it with per field validation, since noticing them at a glance is not a problem anymore

Kudos to @mamazu for their work done in the original PR and the concept as a whole :beers:

### CODE REVIEW

**Comment by mamazu** [comment]
Thanks for taking on the work, I definitely think that this is a great feature that should be implemented.

**Review by jakubtobiasz — APPROVED** [review]

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php [hunk 1]
```diff
@@ -705,6 +706,38 @@ public function iShouldSeeTheActionConfigurationForm()
...
+        $this->updatePage->removeActionFieldValue($channel->getCode(), $field);
+    }
+
+    /**
+     * @When /^I remove the rule amount for ("[^"]+" channel)$/
+     */
+    public function iRemoveTheRuleAmountForChannel(ChannelInterface $channel): void
+    {
+        $this->updatePage->removeRuleAmount($channel->getCode());
+    }
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php]: `When` steps should be defined before `Then` ones

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php [hunk 2]
```diff
@@ -705,6 +706,38 @@ public function iShouldSeeTheActionConfigurationForm()
...
+    /**
+     * @When /^I remove the discount (amount|percentage) for ("[^"]+" channel)$/
+     */
+    public function iRemoveTheDiscountForChannel(string $field, ChannelInterface $channel): void
+    {
+        $this->updatePage->removeActionFieldValue($channel->getCode(), $field);
+    }
+
+    /**
+     * @When /^I remove the rule amount for ("[^"]+" channel)$/
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPromotionsContext.php]:
```suggestion
     * @When I remove the rule amount for :channel channel
```

File: src/Sylius/Behat/Context/Ui/Admin/ManagingShippingMethodsContext.php [hunk 3]
```diff
@@ -520,6 +520,27 @@ public function iAddTheItemsTotalLessThanOrEqualRuleConfiguredWith($value, strin
...
+        );
+    }
+
+    /**
+     * @When /^I remove the shipping charges of ("[^"]+" channel)$/
+     */
+    public function iRemoveTheShippingChargesOfChannel(ChannelInterface $channel): void
+    {
+        $this->updatePage->removeShippingChargesAmount($channel->getCode());
+    }
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingShippingMethodsContext.php]: As above, this method should be moved to be before `Then` ones

File: src/Sylius/Behat/Page/Admin/CatalogPromotion/CreatePage.php [hunk 4]
```diff
@@ -37,7 +37,7 @@ protected function getDefinedElements(): array
             'endDate' => '#sylius_catalog_promotion_endDate',
             'name' => '#sylius_catalog_promotion_name',
             'products' => '[name="sylius_catalog_promotion[scopes][0][configuration][products]"]',
-            'amount' => '[name="sylius_catalog_promotion[actions][0][configuration][amount]"]',
+            'amount' => '[name^="sylius_catalog_promotion[actions][0][configuration]"][name$="[amount]"]',
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/CatalogPromotion/CreatePage.php]: Sort this array in alphabetical order

File: src/Sylius/Behat/Page/Admin/ShippingMethod/UpdatePage.php [hunk 5]
```diff
@@ -54,6 +66,7 @@ protected function getDefinedElements(): array
     {
         return array_merge(parent::getDefinedElements(), [
             'amount' => '#sylius_shipping_method_configuration_%channelCode%_amount',
+            'shipping_charges' => '.ui.segment.configuration',
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/ShippingMethod/UpdatePage.php]: Keys should be in alphabetical order

File: src/Sylius/Behat/Page/Admin/ShippingMethod/CreatePage.php [hunk 6]
```diff
@@ -164,11 +169,23 @@ private function getCollectionItems(string $collection): array
...
+    private function getChannelConfigurationOfLastRule(string $channelCode): NodeElement
     {
         return $this
             ->getLastCollectionItem('rules')
-            ->find('css', sprintf('[id$="configuration"] .field:contains("%s")', $channelName))
+            ->find('css', sprintf('[id^="sylius_shipping_method_rules_"][id$="_configuration_%s"]', $channelCode))
         ;
     }
+
+    private function activateChannelTab(NodeElement $configurationElement, string $channelCode): void
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/ShippingMethod/CreatePage.php]: This function could be extracted to some separated class, as it's already in 3 places 💃

**Review by Zales0123 — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Jan! :tada:

**Pull request merged**